tests/integration.rs

```diff
@@ -19,6 +19,339 @@ use core::time::Duration;
+                    None,
+                )
+                .is_ok()
+            );
+
+            assert!(
+                cert.verify_for_usage(
+                    webpki::ALL_VERIFICATION_ALGS,
+                    &legacy_anchors,
+                    &intermediates_legacy,
+                    time,
+                    KeyUsage::server_auth(),
+                    crls.map(|l| revocation_options_for_test(l)),
+                    None,
+                )
+                .is_ok()
+            );
+
+            let path = cert
+                .verify_for_usage(
+                    webpki::ALL_VERIFICATION_ALGS,
+                    &all_anchors,
+                    &intermediates_legacy,
+                    time,
+                    KeyUsage::server_auth(),
+                    crls.map(|l| revocation_options_for_test(l)),
+                    None,
+                )
+                .unwrap();
+
+            // verify should find shortest path
+            assert!(anchors.contains(path.anchor()));
+        }
+    }
+
+    for &(cert, _dns_name) in &[
+        revoked_demo_root_1,
+        revoked_demo_root_2,
+        revoked_demo_root_3,
+        revoked_demo_root_4,
+    ] {
+        let cert = CertificateDer::from(cert);
+        let cert = webpki::EndEntityCert::try_from(&cert).unwrap();
+
+        for &crls in &[None, Some(&roots_crls)] {
+            assert!(
+                cert.verify_for_usage(
+                    webpki::ALL_VERIFICATION_ALGS,
+                    &anchors,
+                    &intermediates,
+                    time,
+                    KeyUsage::server_auth(),
+                    crls.map(|l| revocation_options_for_test(l)),
+                    None,
+                )
+                .is_ok()
+            );
+        }
+
+        for &crls in &[&intermediates_crls, &all_crls] {
+            assert!(
+                cert.verify_for_usage(
+                    webpki::ALL_VERIFICATION_ALGS,
+                    &anchors,
+                    &intermediates,
+                    time,
+                    KeyUsage::server_auth(),
+                    Some(revocation_options_for_test(crls)),
+                    None,
+                )
+                .is_err_and(|e| matches!(e, webpki::Error::CertRevoked))
+            );
+        }
+    }
+
+    for &(cert, _dns_name) in &[
+        expired_demo_root_1,
+        expired_demo_root_2,
+        expired_demo_root_3,
+        expired_demo_root_4,
+    ] {
+        let cert = CertificateDer::from(cert);
+        let cert = webpki::EndEntityCert::try_from(&cert).unwrap();
+
+        assert!(
+            cert.verify_for_usage(
+                webpki::ALL_VERIFICATION_ALGS,
+                &anchors,
+                &intermediates,
+                time,
+                KeyUsage::server_auth(),
+                None,
+                None,
+            )
+            .is_err_and(|e| matches!(e, webpki::Error::CertExpired { .. }))
+        );
```

Comment:
@cpu @djc if you’re looking for code to trim, this is likely lower value; I included this for completeness (test everything that https://www.amazontrust.com/repository/ has to offer) but there are already tests for expiry.